baybe/campaign.py

```diff
@@ -222,6 +268,100 @@ def add_measurements(
             numerical_measurements_must_be_within_tolerance,
         )
 
+    def _mark_as_measured(
+        self,
+        measurements: pd.DataFrame,
+        numerical_measurements_must_be_within_tolerance: bool,
+    ) -> None:
+        """Mark the given elements of the discrete subspace as measured.
+
+        Args:
+            measurements: A dataframe containing parameter configurations to be
+                marked as measured.
+            numerical_measurements_must_be_within_tolerance: See
+                :func:`baybe.utils.dataframe.fuzzy_row_match`.
+        """
+        idxs_matched = fuzzy_row_match(
+            self.searchspace.discrete.exp_rep,
+            measurements,
+            self.parameters,
+            numerical_measurements_must_be_within_tolerance,
+        )
+        self._searchspace_metadata.loc[idxs_matched, _MEASURED] = True
+
+    @singledispatchmethod
```

Comment:
Can you elaborate why you implemented this via a `singledispatchmethod` instead of doing a check via `isinstance` or similar? For me, the current implementation seems verly complicated. I'd see a use for that if we were to expect many more different types for `constraints`, but I fail to see any futher ones that might make sense.

Reply:
Have changed it since indeed shorter. In the process, I noticed a little typing problem (didn't seem to be an issue in the singledispatch version) that I fixed along the way. Nice side-effect: generalized one function that also occured in another place. Please have a quick look at the corresponding commit and resolve if OK.